src/logging.cpp

```diff
@@ -123,7 +123,7 @@ void logOutputFunction([[maybe_unused]] void* userdata, [[maybe_unused]] int cat
 #ifdef _WIN32
     OutputDebugStringA(output.c_str());
 #endif
-    fmt::fprintf(stderr, output);
+    fmt::fprintf(stderr, "%s", output);
```

Comment:
How about avoiding formatting completely with something like,

```C++
fwrite(output.data(), output.size(), 1, stderr);
```
?

Reply:
The code uses a `fmt::fprintf` rather than the standard C `fprintf`, so I did not want to change that to a C system header function. `fwrite` looks good, there didn't seem to exist a `fmt::fwrite`. Not sure if that matters, or if the C system header `fwrite` would be sufficient - hence kept the semantics as they were.

Reply:
That bug goes back about two years, when I [split](https://github.com/henricj/dunelegacy/commit/7b4bfbdde457b0b8f9695b42b53a9299153aab1a) the existing `fprintf()` into a `fmt::format()` and a call to `fprintf)` (plus, on WIN32, a call to `OutputDebugStringA()`).

The [UCRT](https://devblogs.microsoft.com/cppblog/introducing-the-universal-crt/) `vfprintf()` called from the AdPlug library's debug code results in one WriteFile kernel call for each character in the string.
![Screenshot 2022-06-17 151925](https://user-images.githubusercontent.com/4596789/174408889-4910ba76-355d-4c78-8365-861d145804dc.png)

This behavior is probably some backwards compatibility thing for when the output isn't fully buffered (and UCRT doesn't support line buffering).  Anyway, a call to `fwrite()` should result in single `WriteFile` kernel call of the same size as passed to `fwrite()`.

Reply:
Gotcha - updated to use fwrite.